[manage data] Add Data tiers content (#429)
elastic/docs-content: manage-data/lifecycle/data-tiers.md

## Configure data tiers [configure-data-tiers]

Follow the instructions for your deployment type to configure data tiers.

### {{ess}} or {{ece}} [configure-data-tiers-cloud]

The default configuration for an {{ecloud}} deployment includes a shared tier for hot and content data. This tier is required and can’t be removed.

#### Add a data tier

To add a warm, cold, or frozen tier when you create a deployment:

1. On the **Create deployment** page, click **Advanced Settings**.
2. Click **+ Add capacity** for any data tiers to add.
3. Click **Create deployment** at the bottom of the page to save your changes.

:::{image} ../../images/elasticsearch-reference-ess-advanced-config-data-tiers.png
:alt: {{ecloud}}'s deployment Advanced configuration page
:class: screenshot
:::

To add a data tier to an existing deployment:

> 1. Log in to the [{{ecloud}} console](https://cloud.elastic.co?page=docs&placement=docs-body).

Reviewer: ```suggestion
1. Log in to the [{{ecloud}} console](https://cloud.elastic.co?page=docs&placement=docs-body) or the {{ece}} admin console.
```

ECE users have a different log in location. The term "Elastic Cloud" also probably needs to change, but I think that's a problem for another day. :-)

Author: I'm going to use tabs for this!

Author: Oh wait never mind! I will use your suggestion.

2. On the **Deployments** page, select your deployment.
3. In your deployment menu, select **Edit**.
4. Click **+ Add capacity** for any data tiers to add.
5. Click **Save** at the bottom of the page to save your changes.

#### Disable a data tier [disable-a-data-tier]

The process of disabling a data tier depends on whether we are dealing with [searchable snapshots](#ece-disable-searchable-snapshot-data-tier) or [regular indices](#ece-disable-non-searchable-snapshot-data-tier).

The hot and warm tiers store regular indices, while the frozen tier stores searchable snapshots. However, the cold tier can store either regular indices or searchable snapshots. To check if a cold tier contains searchable snapshots perform the following request:

```sh
GET /_cat/indices/restored-*
```

##### Non-searchable snapshot data tier [ece-disable-non-searchable-snapshot-data-tier]

> Elastic Cloud Enterprise tries to move all data from the nodes that are removed during plan changes. To disable a non-searchable snapshot data tier (e.g., hot, warm, or cold tier), make sure that all data on that tier can be re-allocated by reconfiguring the relevant shard allocation filters. You’ll also need to temporarily stop your index lifecycle management (ILM) policies to prevent new indices from being moved to the data tier you want to disable.

Reviewer: ```suggestion
{{ece}} tries to move all data from the nodes that are removed during plan changes. To disable a non-searchable snapshot data tier (e.g., hot, warm, or cold tier), make sure that all data on that tier can be re-allocated by reconfiguring the relevant shard allocation filters. You’ll also need to temporarily stop your index lifecycle management (ILM) policies to prevent new indices from being moved to the data tier you want to disable.
```

Author: Earlier on the page we use `{{ess}} or {{ece}}`. Should I replace all mentions of `{{ess}}` with `{{ech}}` or use `{{ess}} and {{ece}}` here? I'd like to keep it consistent.

Reviewer: I would use `{{ech}}` (Elastic Cloud Hosted). Apparently that term is not an exact replacement for `{{ess}}` (Elasticsearch Service), but I think it's the best we have right now since the latter is definitely depricated.

> To learn more about ILM for Elastic Cloud Enterprise, or shard allocation filtering, check the following documentation:

Reviewer: ```suggestion
To learn more about ILM for {{ece}}, or shard allocation filtering, check the following documentation:
```
I suppose we may fix all these entities automagically at some point, so feel free to ignore.

> * [Configure index management](https://www.elastic.co/guide/en/cloud-enterprise/current/ece-configure-index-management.html)

Author: I checked the IA inventory and it looks like this page is not being migrated because "Index management is deprecated functionality. Users should use ILM instead." So I'm going to remove it instead.

* [Create your index lifecyle policy](https://www.elastic.co/guide/en/elasticsearch/reference/current/set-up-lifecycle-policy.html)
* [Managing the index lifecycle](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-lifecycle-management.html)
* [Shard allocation filters](https://www.elastic.co/guide/en/elasticsearch/reference/current/shard-allocation-filtering.html)

To make sure that all data can be migrated from the data tier you want to disable, follow these steps:

1. Determine which nodes will be removed from the cluster.

>     1. [Log into the Cloud UI](/deploy-manage/deploy/cloud-enterprise/log-into-cloud-ui.md).

Author: I'm going to use tabs for this.

Alternatively, you can use the cold tier to store regular indices with replicas instead of using {{search-snaps}}. This lets you store older data on less expensive hardware but doesn’t reduce required disk space compared to the warm tier.

### Frozen tier [frozen-tier]

Once data is no longer being queried, or being queried rarely, it may move from the cold tier to the frozen tier where it stays for the rest of its life.

The frozen tier requires a snapshot repository. The frozen tier uses [partially mounted indices](/deploy-manage/tools/snapshot-and-restore/searchable-snapshots.md#partially-mounted) to store and load data from a snapshot repository. This reduces local storage and operating costs while still letting you search frozen data. Because {{es}} must sometimes fetch frozen data from the snapshot repository, searches on the frozen tier are typically slower than on the cold tier.

## Configure data tiers [configure-data-tiers]

Follow the instructions for your deployment type to configure data tiers.

> ### {{ess}} or {{ece}} [configure-data-tiers-cloud]

Author: ```suggestion
### {{ech}} or {{ece}} [configure-data-tiers-cloud]
```

:alt: {{ecloud}}'s deployment Advanced configuration page
:class: screenshot
:::

To add a data tier to an existing deployment:

1. Log in to the [{{ecloud}} console](https://cloud.elastic.co?page=docs&placement=docs-body) or the {{ece}} admin console.
2. On the **Deployments** page, select your deployment.
3. In your deployment menu, select **Edit**.
4. Click **+ Add capacity** for any data tiers to add.
5. Click **Save** at the bottom of the page to save your changes.

#### Disable a data tier [disable-a-data-tier]

The process of disabling a data tier depends on whether we are dealing with [searchable snapshots](#ece-disable-searchable-snapshot-data-tier) or [regular indices](#ece-disable-non-searchable-snapshot-data-tier).

The hot and warm tiers store regular indices, while the frozen tier stores searchable snapshots. However, the cold tier can store either regular indices or searchable snapshots. To check if a cold tier contains searchable snapshots perform the following request:

```sh
GET /_cat/indices/restored-*
```

##### Non-searchable snapshot data tier [ece-disable-non-searchable-snapshot-data-tier]

> {{ess}} and {{ece}} try to move all data from the nodes that are removed during plan changes. To disable a non-searchable snapshot data tier (e.g., hot, warm, or cold tier), make sure that all data on that tier can be re-allocated by reconfiguring the relevant shard allocation filters. You’ll also need to temporarily stop your index lifecycle management (ILM) policies to prevent new indices from being moved to the data tier you want to disable.

Author: ```suggestion
{{ech}} and {{ece}} try to move all data from the nodes that are removed during plan changes. To disable a non-searchable snapshot data tier (e.g., hot, warm, or cold tier), make sure that all data on that tier can be re-allocated by reconfiguring the relevant shard allocation filters. You’ll also need to temporarily stop your index lifecycle management (ILM) policies to prevent new indices from being moved to the data tier you want to disable.
```

To learn more about ILM, or shard allocation filtering, check the following documentation:

* [Create your index lifecyle policy](/manage-data/lifecycle/index-lifecycle-management/configure-lifecycle-policy.md)
* [Managing the index lifecycle](/manage-data/lifecycle/index-lifecycle-management.md)
* [Shard allocation filters](/deploy-manage/distributed-architecture/shard-allocation-relocation-recovery/index-level-shard-allocation.md)

To make sure that all data can be migrated from the data tier you want to disable, follow these steps:

1. Determine which nodes will be removed from the cluster.

    :::::{tab-set}

>     ::::{tab-item} {{ess}}

Author: ```suggestion
    ::::{tab-item} {{ech}}
```
